src/shared/dlt_common.c

```diff
@@ -187,7 +188,8 @@ DltReturnValue dlt_print_mixed_string(char *text, int textlength ,uint8_t *ptr,
     for (lines=0; lines< (size / DLT_COMMON_HEX_CHARS); lines++)
     {
         /* Line number */
-        snprintf(text,DLT_COMMON_HEX_LINELEN+1,"%.6x: ",lines * DLT_COMMON_HEX_CHARS);
+        line = (lines * DLT_COMMON_HEX_CHARS) & 0x00FFFFFF;
```

Comment:
Interesting. I have the opposite preference here. The purpose seems to be to limit the value to 3 byte size (or 6 hex chars, like before?)
Of course it can be discussed if such bit manipulation should be done at all, but IF it is done in my opinion I will much easier understand it by seeing the value. This is for me a special case for masking, bitshifting and such things, where a define would just hide the information that is important here. Obviously other magic values should often be named constants (or defines) of course.

(The problem lies more in doing low level manipulation at all, but when you do that, make it explicit IMHO. Another way to avoid repetition might  be some refactoring involving a common function call maybe)

Just my opinion.

Reply:
changed